holidays/countries/thailand.py

```diff
@@ -649,11 +736,11 @@ def _add_observed(dt: date) -> None:
             2023: (MAY, 11),
         }
         # For years with exact date data.
-        if year in raeknakhwan_dates:
-            _add_observed(self._add_holiday(raeknakhwan, *raeknakhwan_dates[year]))
+        if self._year in raeknakhwan_dates:
+            self._add_holiday(raeknakhwan, *raeknakhwan_dates[self._year])
         # Approx. otherwise for 1957-2013.
-        elif 1957 <= year <= 1996:
-            _add_observed(self._add_holiday(raeknakhwan, MAY, 13))
+        elif 1957 <= self._year <= 1996:
+            self._add_holiday(raeknakhwan, MAY, 13)
```

Comment:
Hmm, I'ld say no to this for now since the date chosen by the Royal Palace's astrologer is quite erratic as of late as of late, but will keep this in mind :eyes:

Reply:
We can write:
```python
        if 1957 <= self._year <= 2023 and self._year != 1999:
            # Royal Ploughing Ceremony.
            self._add_holiday(tr("วันพืชมงคล"), *raeknakhwan_dates.get(self._year, (MAY, 13)))
```
and change number every time (obviously every year) when adding a new date to `raeknakhwan_dates`.

Reply:
Fixed.